src/System.Private.CoreLib/shared/System/Globalization/TimeSpanFormat.cs

```diff
@@ -69,45 +106,35 @@ internal static bool TryFormat(TimeSpan value, Span<char> destination, out int c
             return false;
         }
 
-        private static StringBuilder FormatToBuilder(TimeSpan value, ReadOnlySpan<char> format, IFormatProvider formatProvider)
+        internal static string FormatC(TimeSpan value)
         {
-            // Standard formats other than 'c'/'t'/'T', which should have already been handled.
-            if (format.Length == 1)
-            {
-                char f = format[0];
-                switch (f)
-                {
-                    case 'g':
-                    case 'G':
-                        DateTimeFormatInfo dtfi = DateTimeFormatInfo.GetInstance(formatProvider);
-                        return FormatG(
-                            value, 
-                            format: value.Ticks < 0 ? dtfi.FullTimeSpanNegativePattern : dtfi.FullTimeSpanPositivePattern,
-                            full: f == 'G');
-
-                    default:
-                        throw new FormatException(SR.Format_InvalidString);
-                }
-            }
-
-            // Custom formats
-            return FormatCustomized(value, format, DateTimeFormatInfo.GetInstance(formatProvider), result: null);
+            Span<char> destination = stackalloc char[26]; // large enough for any "c" TimeSpan
+            TryFormatStandard(value, StandardTimeSpanFormat.C, null, destination, out int charsWritten);
+            return new string(destination.Slice(0, charsWritten));
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool IsFormatC(ReadOnlySpan<char> format) =>
-            format.Length == 0 ||
-            (format.Length == 1 && (format[0] == 'c' || (format[0] | 0x20) == 't'));
-
-        internal static string FormatC(TimeSpan value)
+        private static string FormatG(TimeSpan value, DateTimeFormatInfo dtfi, StandardTimeSpanFormat format)
         {
-            Span<char> destination = stackalloc char[26]; // large enough for any "c" TimeSpan
-            TryFormatC(value, destination, out int charsWritten);
+            string decimalSeparator = dtfi.DecimalSeparator;
+            int maxLength = 25 + decimalSeparator.Length; // large enough for any "g"/"G" TimeSpan
+            Span<char> destination = maxLength < 128 ?
+                stackalloc char[maxLength] :
+                new char[maxLength]; // the chances of needing this case are almost 0, as DecimalSeparator.Length will basically always == 1
+            TryFormatStandard(value, format, dtfi.DecimalSeparator, destination, out int charsWritten);
             return new string(destination.Slice(0, charsWritten));
         }
 
-        private static bool TryFormatC(TimeSpan value, Span<char> destination, out int charsWritten)
+        private enum StandardTimeSpanFormat
```

Comment:
Is it standard practice to explicitly set the non-flag enum values rather than let the compiler set them? I don't see their ordinal values being used.

Reply:
Will remove.